.github/workflows/build-deploy-dev.yml

```diff
@@ -197,23 +196,27 @@ jobs:
       - name: Log in to OpenShift
         run: |
           oc login --token=${{ secrets.SA_TOKEN }} --server=https://api.silver.devops.gov.bc.ca:6443
-      # Deploy SIMS-Api
-      - name: Deploy SIMS-API
-        working-directory: "./devops/"
-        run: |
-          make oc-deploy-api
-      # Deploy Workers
-      - name: Deploy Workers
+          if [ $? -ne 0 ]; then
+            echo "Failed to log in to OpenShift. Please check the openshift credentials and connection details."
+            exit 1
+          fi
+      # Run db-migrations
+      - name: Run db-migrations
         working-directory: "./devops/"
         run: |
-          make oc-deploy-workers
-      # Deploy Queue Consumers
-      - name: Deploy Queue Consumers
+          make oc-run-db-migrations
+          if [ $? -ne 0 ]; then
+            echo "Failed to run DB Migrations."
+            exit 1
+          fi
+      # Deploy SIMS-API, Workers, Queues, and Web/Frontend in parallel after db-migrations.
+      - name: Deploy SIMS-API, Workers, Queues, and Web/Frontend in parallel
         working-directory: "./devops/"
         run: |
-          make oc-deploy-queue-consumers
-      # Deploy Deploy Web/Frontend
-      - name: Deploy Web/Frontend
-        working-directory: "./devops/"
+          make oc-deploy-api &
+          make oc-deploy-workers &
+          make oc-deploy-queue-consumers &
+          make oc-deploy-web &
+      - name: Wait for parallel deployments to finish
         run: |
-          make oc-deploy-web
+          wait
```

Comment:
will it wait for all processes to finish or it wait for a certain time? how is it for my info

Reply:
Not for certain time, once the parallel tasks are executed, it just waits like promise.all